apm-agent-plugins/apm-mongodb/apm-mongodb-common/src/test/java/co/elastic/apm/agent/mongodb/AbstractMongoClientInstrumentationIT.java

```diff
@@ -70,10 +76,12 @@ public void endTransaction() throws Exception {
             currentTransaction.deactivate().end();
         }
 
-        // drop outside of transaction to prevent creating any span
-        dropCollection();
-
-        reporter.reset();
+        try {
+            // drop outside of transaction to prevent creating any span
+            dropCollection();
```

Comment:
[for reviewer] this might throw an exception when the test did not create a collection, the try/finally allows to make sure that we always reset the reporter.